### ISSUE TITLE
Help - A Route with two possible states

### ISSUE THREAD

**Issue by sampurcell93** [rackt/react-router#2145]
Hi there! I am building an application where one route can have two states (hear me out!). This route is defined at /search/:query. There are two ways to access it. The first is by entering the search term into the search bar, at which point all of the current route's content blurs, and the search page appears on top of it. To do this, I need to render the search page on top of the current page, *without* replacing it.

The other way to access the page is to simply refresh /search/:query at which point the search results will appear on a plain white background. I think a good analogy is the facebook image gallery. If you access the gallery while on the website, it pops over as a darkened image gallery. But if you refresh the page, it's a separate page.

I have managed to make this work (sort of)! My problem is that in the first case, when I am done with the blurred-popover search results page, and I click a `<Link/>` to a new route, that route's onEnter function is not called, and I don't know a better way to ensure that the blur effect is removed and the popover page is removed.

In short, I am currently looking for a way to navigate to a route without triggering a whole-page replace, but still letting the router know that we have indeed navigated, so that navigating away to another page triggers onEnter functions. I am currently using router.replaceWith() to do the replace, but I'm a little stumped. Someone has pointed me in the direction of the rackt/history project, but I'm still a bit lost.

Thanks in advance!

**Comment by knowbody** [comment]
this is exactly what [pinterest example](https://github.com/rackt/react-router/tree/master/examples/pinterest) does.
